xlators/meta/src/meta-helpers.c

```diff
@@ -42,20 +37,17 @@ meta_fd_get(fd_t *fd, xlator_t *this)
 int
 meta_fd_release(fd_t *fd, xlator_t *this)
 {
-    uint64_t value = 0;
     meta_fd_t *meta_fd = NULL;
     int i = 0;
 
-    fd_ctx_get(fd, this, &value);
-    meta_fd = (void *)(uintptr_t)value;
-
-    if (meta_fd && meta_fd->dirents) {
-        for (i = 0; i < meta_fd->size; i++)
-            GF_FREE((void *)meta_fd->dirents[i].name);
-        GF_FREE(meta_fd->dirents);
-    }
+    meta_fd = fd_ctx_get_ptr(fd, this);
```

Comment:
Probably not a critical bug because we are already destroying the fd if we are here, but we should remove the ctx from the fd if we are going to destroy it.

```suggestion
    meta_fd = fd_ctx_del_ptr(fd, this);
```